pkg/rpc/server/prometheus.go

```diff
@@ -1,249 +1,63 @@
-			Namespace: "neogo",
-		},
-	)
-
-	getrawmempoolCalled = prometheus.NewCounter(
-		prometheus.CounterOpts{
-			Help:      "Number of calls to getrawmempool rpc endpoint",
-			Name:      "getrawmempool_called",
-			Namespace: "neogo",
-		},
-	)
-
-	validateaddressCalled = prometheus.NewCounter(
-		prometheus.CounterOpts{
-			Help:      "Number of calls to validateaddress rpc endpoint",
-			Name:      "validateaddress_called",
-			Namespace: "neogo",
-		},
-	)
-
-	getassetstateCalled = prometheus.NewCounter(
-		prometheus.CounterOpts{
-			Help:      "Number of calls to getassetstate rpc endpoint",
-			Name:      "getassetstate_called",
-			Namespace: "neogo",
-		},
-	)
-
-	getaccountstateCalled = prometheus.NewCounter(
-		prometheus.CounterOpts{
-			Help:      "Number of calls to getaccountstate rpc endpoint",
-			Name:      "getaccountstate_called",
-			Namespace: "neogo",
-		},
-	)
-
-	gettransactionheightCalled = prometheus.NewCounter(
-		prometheus.CounterOpts{
-			Help:      "Number of calls to gettransactionheight rpc endpoint",
-			Name:      "gettransactionheight_called",
-			Namespace: "neogo",
-		},
-	)
-
-	gettxoutCalled = prometheus.NewCounter(
-		prometheus.CounterOpts{
-			Help:      "Number of calls to gettxout rpc endpoint",
-			Name:      "gettxout_called",
-			Namespace: "neogo",
-		},
-	)
-
-	getrawtransactionCalled = prometheus.NewCounter(
-		prometheus.CounterOpts{
-			Help:      "Number of calls to getrawtransaction rpc endpoint",
-			Name:      "getrawtransaction_called",
-			Namespace: "neogo",
-		},
-	)
-
-	getunclaimedCalled = prometheus.NewCounter(
-		prometheus.CounterOpts{
-			Help:      "Number of calls to getunclaimed rpc endpoint",
-			Name:      "getunclaimed_called",
-			Namespace: "neogo",
-		},
-	)
-
-	getunspentsCalled = prometheus.NewCounter(
-		prometheus.CounterOpts{
-			Help:      "Number of calls to getunspents rpc endpoint",
-			Name:      "getunspents_called",
-			Namespace: "neogo",
-		},
-	)
-
-	sendrawtransactionCalled = prometheus.NewCounter(
-		prometheus.CounterOpts{
-			Help:      "Number of calls to sendrawtransaction rpc endpoint",
-			Name:      "sendrawtransaction_called",
-			Namespace: "neogo",
-		},
-	)
-
-	submitblockCalled = prometheus.NewCounter(
-		prometheus.CounterOpts{
-			Help:      "Number of calls to submitblock rpc endpoint",
-			Name:      "submitblock_called",
-			Namespace: "neogo",
-		},
-	)
-
-	getstorageCalled = prometheus.NewCounter(
-		prometheus.CounterOpts{
-			Help:      "Number of calls to getstorage rpc endpoint",
-			Name:      "getstorage_called",
-			Namespace: "neogo",
-		},
-	)
+	rpcCalls = []string{
+		"getapplicationlog",
+		"getbestblockhash",
+		"getbestblock",
+		"getblockcount",
+		"getblockhash",
+		"getblockheader",
+		"getblocksysfee",
+		"getclaimable",
+		"getconnectioncount",
+		"getcontractstate",
+		"getvalidators",
+		"getnep5balances",
+		"getnep5transfers",
+		"getversion",
+		"getpeers",
+		"getrawmempool",
+		"validateaddress",
```

Comment:
Not a sorted list. And I think adding `invoke*` there is also useful, every call should have a metric associated.

Reply:
Done!